pkg/controller/accountclaim/reuse_test.go

```diff
@@ -0,0 +1,165 @@
+
+		case errMsg = <-errors:
+
+		}
+	}()
+	err := functorun(testutils.NullLogger{}, client, notifications, errors)
+
+	return msg, errMsg, err
+}
+
+var _ = Describe("Account Reuse", func() {
+	var (
+		r                *accountclaim.ReconcileAccountClaim
+		ctrl             *gomock.Controller
+		awsClientBuilder *awsmock.Builder
+		mockAwsClient    *awsmock.MockClient
+		kubeClient       kclient.Client
+	)
+
+	BeforeEach(func() {
+		ctrl = gomock.NewController(GinkgoT())
+		awsClientBuilder = &awsmock.Builder{MockController: ctrl}
+		kubeClient = mock.NewMockClient(ctrl)
+
+		// Create the reconciler with a mocking AWS client IBuilder.
+		r = accountclaim.NewReconcileAccountClaim(
+			kubeClient,
+			scheme.Scheme,
+			awsClientBuilder,
+		)
+
+		mockAwsClient = awsmock.NewMockClient(ctrl)
+	})
+
+	AfterEach(func() {
+		ctrl.Finish()
+	})
+
+	Describe("CleanUpAwsAccountVpcEndpointServiceConfigurations", func() {
+		var (
+			describeOutput ec2.DescribeVpcEndpointServiceConfigurationsOutput
+			deleteOutput   ec2.DeleteVpcEndpointServiceConfigurationsOutput
+		)
+		Context("When no VPC Endpoint Service Configuration exists", func() {
+			BeforeEach(func() {
+				mockAwsClient.EXPECT().DescribeVpcEndpointServiceConfigurations(gomock.Any()).Return(&describeOutput, nil)
+			})
+
+			It("Does nothing", func() {
+				notifications, errors, _ := runCleanupFunc(r.CleanUpAwsAccountVpcEndpointServiceConfigurations, mockAwsClient)
+				Expect(errors).To(Equal(""))
+				Expect(notifications).To(Equal("VPC endpoint service configuration cleanup finished successfully (nothing to do)"))
+			})
+		})
+
+		Context("When one VPC Endpoint Service Configuration exists", func() {
+			var (
+				serviceConfigId string
+				deleteInput     *ec2.DeleteVpcEndpointServiceConfigurationsInput
+			)
+			BeforeEach(func() {
+				serviceConfigId = "FakeID"
+				describeOutput = ec2.DescribeVpcEndpointServiceConfigurationsOutput{
+					ServiceConfigurations: []*ec2.ServiceConfiguration{
+						{
+							ServiceId: &serviceConfigId,
+						},
+					},
+				}
+				deleteOutput = ec2.DeleteVpcEndpointServiceConfigurationsOutput{}
+				mockAwsClient.EXPECT().DescribeVpcEndpointServiceConfigurations(gomock.Any()).Return(&describeOutput, nil)
+				mockAwsClient.EXPECT().DeleteVpcEndpointServiceConfigurations(gomock.Any()).Do(func(input *ec2.DeleteVpcEndpointServiceConfigurationsInput) {
+					deleteInput = input
+				}).Return(&deleteOutput, nil)
+			})
+
+			It("Deletes the VPC Endpoint Service Configuration", func() {
+				notifications, errors, _ := runCleanupFunc(r.CleanUpAwsAccountVpcEndpointServiceConfigurations, mockAwsClient)
+
+				Expect(len(deleteInput.ServiceIds)).To(Equal(1))
+				Expect(*deleteInput.ServiceIds[0]).To(Equal(serviceConfigId))
+				Expect(errors).To(Equal(""))
+				Expect(notifications).To(Equal("VPC endpoint service configuration cleanup finished successfully"))
+			})
+		})
+
+		Context("When two VPC Endpoint Service Configuration exist", func() {
```

Comment:
Just missing 1 other test where we cover if one of the VPC enpoint deletions returns an error.

Reply:
done